tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/Game.kt

```diff
@@ -0,0 +1,93 @@
+package com.anaplan.engineering.azuki.tictactoe.implementation
+
+import org.slf4j.LoggerFactory
+
+class Game(
+    val board: MutableList<MutableList<Token?>>,
+    val playOrder: PlayOrder,
+) {
+
+    val width by lazy { board[0].size }
+    val height by lazy { board.size }
+
+    private val lines by lazy {
+        mutableListOf<List<Pair<Int, Int>>>().apply {
+            addAll((0 until height).map { y -> (0 until width).map { x -> y to x } })
+            addAll((0 until width).map { x -> (0 until height).map { y -> y to x } })
+            if (width == height) {
+                add((0 until width).map { i -> i to i })
+                add((0 until width).map { i -> width - 1 - i to i })
+            }
```

Comment:
⛏️ Not fundamental right now since the adapter is only creating 3x3 games, but "full-length" diagonals should also be considered for rectangular games.

Reply:
a) the spec is 3x3
b) the spec defines the rules for diagonals
c) tictactoe is by definition exactly 3x3

so no other diagonals should not be considered, not should it be possible for width and height to vary